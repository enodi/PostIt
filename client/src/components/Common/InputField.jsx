import React from 'react';
import PropTypes from 'prop-types';

const InputField = ({
  name, value, id, type, htmlFor, label, onChange, onBlur, onFocus, placeholder
}) => (
  <div className="input-field col s12" >
    <input
      value={value}
      onChange={onChange}
      type={type}
      onBlur={onBlur}
      onFocus={onFocus}
      name={name}
      id={id}
      placeholder={placeholder}
    />
    <label htmlFor={htmlFor} > {label} </label>
  </div >
);

InputField.PropTypes = {
  name: PropTypes.string.isRequired,
  value: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  type: PropTypes.string.isRequired,
  htmlFor: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired,
  id: PropTypes.string.isRequired,
  onBlur: PropTypes.func.isRequired,
  onFocus: PropTypes.func.isRequired,
  placeholder: PropTypes.string.isRequired
};

InputField.defaultProps = {
  type: 'text'
};

export default InputField;
