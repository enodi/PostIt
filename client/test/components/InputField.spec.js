import React from 'react';
import { shallow } from 'enzyme';

import InputField from '../../src/components/Common/InputField.jsx';

describe('InputField Component', () => {
  it('should render InputField Component', () => {
    const wrapper = shallow(<InputField />);
    expect(wrapper).toMatchSnapshot();
  });
});
