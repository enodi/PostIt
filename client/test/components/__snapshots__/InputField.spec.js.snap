// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputField Component should render InputField Component 1`] = `
<div
  className="input-field"
>
  <input
    type="text"
  />
  <label>
     
     
  </label>
</div>
`;
